### PR TITLE
Have bpfmt show usage when run without any arguments

### PR DESCRIPTION
**Pull request by mathjeff**

### CODE REVIEW

File: bpfmt/bpfmt.go [hunk 1]
```diff
@@ -40,17 +41,18 @@ func usage() {
...
-// If in == nil, the source is the contents of the file with the given filename.
-func processFile(filename string, in io.Reader, out io.Writer) error {
-	if in == nil {
-		f, err := os.Open(filename)
-		if err != nil {
-			return err
-		}
-		defer f.Close()
-		in = f
+func processFile(filename string, out io.Writer) error {
```
Comment (mathjeff) [inline bpfmt/bpfmt.go]: it turns out that this patch ("Refactor in preparation to detect when stdout is the input file") didn't end up being needed, but I kept it in anyway thinking that it was slightly cleaner

File: bpfmt/bpfmt.go [hunk 2]
```diff
@@ -40,17 +41,18 @@ func usage() {
...
+	f, err := os.Open(filename)
+	if err != nil {
+		return err
 	}
+	defer f.Close()
+	in := f
```
Comment (colincross) [inline bpfmt/bpfmt.go]: This doesn't make sense any more, either rename f to in or pass f to processReader

File: bpfmt/bpfmt.go [hunk 3]
```diff
@@ -36,8 +37,13 @@ func report(err error) {
 }
 
 func usage() {
+	usageViolation("")
+}
+
+func usageViolation(violation string) {
 	fmt.Fprintf(os.Stderr, "usage: bpfmt [flags] [path ...]\n")
 	flag.PrintDefaults()
+	fmt.Fprintln(os.Stderr, violation)
```
Comment (colincross) [inline bpfmt/bpfmt.go]: I think this traditionally is printed first before usage:

Comment (mathjeff) [reply bpfmt/bpfmt.go]: done

File: bpfmt/bpfmt.go (hunk 2 above)
Comment (mathjeff) [reply bpfmt/bpfmt.go]: thanks

**Pull request merged**